xm.js

```diff
@@ -279,26 +294,24 @@ function RedrawScreen() {
   }
 }
 
-var cur_songpos = -1, cur_pat = -1, cur_row = 64, cur_ticksamp = 0;
-var cur_tick = 6;
 function next_row() {
-  if (cur_pat == -1 || cur_row >= xm.patterns[cur_pat].length) {
-    cur_row = 0;
-    cur_songpos++;
-    if (cur_songpos >= xm.songpats.length)
-      cur_songpos = xm.song_looppos;
-    cur_pat = xm.songpats[cur_songpos];
+  if (player.cur_pat == -1 || player.cur_row >= player.xm.patterns[player.cur_pat].length) {
+    player.cur_row = 0;
+    player.cur_songpos += 1;
```

Comment:
Ah, nah I just brainfarted and thought I couldn't ++ fields on objects. But that's not true at all :) I'll fix it!